toolkit/components/uniffi-bindgen-gecko-js/src/interface_ext.rs

```diff
@@ -222,3 +226,118 @@ pub impl FFIArgument {
         matches!(self.type_(), FFIType::RustBuffer)
     }
 }
+
```

Comment:
I'm starting to be not a great fan of extending everything 😕 But it got the job done pretty quickly - happy to re-shape the code here though

Reply:
Interested to know what you see as the downsides here.  I really want to find a nice way to express this code, both for these bindings and in general.

Reply:
My only downside is seeing a lot of `js_*` functions and `cpp_*`, etc... it's slightly frustrating and I imagine there is a cleaner way? Regardless it's not something we should worry about *now* - maybe I'll add a small comment that it's not the prettiest?